include/envoy/http/filter.h

```diff
@@ -119,6 +119,42 @@ enum class FilterMetadataStatus {
   Continue,
 };
 
+/**
+ * An interface for a per-connection object that is created by a filter and then managed by the
+ * HTTP connection manager. This pushes singleton handling into the HTTP connection manager.
+ */
+
+class MutableHttpConnection;
+
+class PerConnectionObject {
+public:
+  virtual ~PerConnectionObject() = default;
+
+  /**
+   * @return MutableHttpConnection& the mutable http connection, this is expected not to be nullptr.
+   */
+  virtual MutableHttpConnection& mutable_http_connection() PURE;
```

Comment:
Is this accessor really needed in practice? Since the mutable http connection is passed to the object construction, each per connection object can store it if it wishes, or not at all if it's not needed. So I think I would remove this.

Reply:
Interpretation here is now clear to me and will remove this.